api/src/main/java/jakarta/data/metamodel/TextAttribute.java

```diff
@@ -33,11 +35,84 @@ public interface TextAttribute<T> extends SortableAttribute<T> {
      */
     Sort<T> ascIgnoreCase();
 
+    default TextRestriction<T> contains(String substring) {
+        return Restrict.contains(substring, name());
+    }
+
     /**
      * Obtain a request for a descending, case insensitive {@link Sort} based on the entity attribute.
      *
      * @return a request for a descending, case insensitive sort on the entity attribute.
      */
     Sort<T> descIgnoreCase();
 
+    default TextRestriction<T> endsWith(String suffix) {
+        return Restrict.endsWith(suffix, name());
+    }
+
+    default TextRestriction<T> equalTo(String value) {
+        return Restrict.equalTo(value, name());
+    }
+
+    default TextRestriction<T> greaterThan(String value) {
+        return Restrict.greaterThan(value, name());
+    }
+
+    default TextRestriction<T> greaterThanEqual(String value) {
+        return Restrict.greaterThanEqual(value, name());
+    }
+
+    default TextRestriction<T> lessThan(String value) {
+        return Restrict.lessThan(value, name());
+    }
+
+    default TextRestriction<T> lessThanEqual(String value) {
+        return Restrict.lessThanEqual(value, name());
+    }
+
+    // TODO once we have Pattern:
+    //default TextRestriction<T> like(Pattern pattern) {
+    //    return Restrict.like(pattern, name());
+    //}
+
+    default TextRestriction<T> like(String pattern) {
+        return Restrict.like(pattern, name());
+    }
+
+    default TextRestriction<T> like(String pattern,
+                                    char charWildcard,
+                                    char stringWildcard) {
+        return Restrict.like(pattern, charWildcard, stringWildcard, name());
+    }
+
+    default TextRestriction<T> not(String value) {
```

Comment:
The same point of using the Java core as reference to us:

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/function/Predicate.html